_posts/2024-12-17-release-0_9_0.md

```diff
@@ -0,0 +1,32 @@
+---
+layout: post
+title:  "kroxylicious release 0.9.0"
+date:   2024-12-17 00:00:00 +0000
+author: "Sam Barker"
+author_url: "https://github.com/SamBarker"
+categories:  [releases, kroxylicious, kafka3.9]
+---
+
+The Kroxylicious project is very pleased to announce the [0.9.0](https://github.com/kroxylicious/kroxylicious/releases/tag/v0.9.0) release of Kroxylicious. This release introduces support for Apache Kafka 3.9. 
+
+We generally expect Kroxylicious to be both forwards and backwards compatible across Kafka version just like the Apache Kafka client and broker. While we tested with pre-release versions Kafka 3.9 we discovered after it was released that we had missed a case in the protocol version negotiation. When the API Versions request version was updated we were unable to decode the response. In the 0.9.0 release we adopt the same behaviour as the Kafka broker and respond with API version 0 if we do not support the protocol version requested by the client - we have also expanded our test coverage to get earlier warnings of similar issues in the future. 
+
+This release also marks the introduction of our new connection handling sate machine at the core of the proxy. We have evolved state machine at the core of the proxy to separate out the state carried at various stages of the connection life cycle and provide stronger rules around transitions between states. This is still a work in progress as its implications are rather wide-ranging however with think it gives us a stronger core to base everything else around. This does for the moment limit our ability to offer authentication offload support, we do intend to restore this eventually (please do get in touch if this use case is important to you).
+
+There are many quality of life improvements in this release (in no particular order)
+
+- Ensure we now respond with errors instead of closing the connection when there are problems encrypting records 
```

Comment:
Would be good to link to the relevant issue or PR for each of these, for example:
```suggestion
- Ensure we now respond with errors instead of closing the connection when there are problems encrypting records ([#1630](https://github.com/kroxylicious/kroxylicious/pull/1630))
```